tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/interceptor/FaultToleranceInterceptorTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.fault.tolerance.tck.interceptor;
+
+
+import org.eclipse.microprofile.fault.tolerance.tck.interceptor.CounterFactory.CounterId;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.testng.Assert.fail;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * A Client to demonstrate the interceptors ordering behavior of FT and CDI annotations
+ * @author carlosdlr
+ */
+public class FaultToleranceInterceptorTest extends Arquillian {
+
+    @Inject
+    @CounterId("EarlyFtInterceptor")
+    private AtomicInteger earlyInterceptorCounter;
+
+    @Inject
+    @CounterId("LateFtInterceptor")
+    private AtomicInteger lateInterceptorCounter;
+
+    @Inject
+    @CounterId("serviceA")
+    private AtomicInteger methodCounter;
+
+    @Inject
+    private InterceptorComponent testInterceptor;
+
+    @Deployment
+    public static WebArchive deploy() {
+        JavaArchive testJar = ShrinkWrap
+            .create(JavaArchive.class, "interceptorFtCdi.jar")
+            .addClasses(InterceptorComponent.class, EarlyFtInterceptor.class, LateFtInterceptor.class, CounterFactory.class)
+             .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
+            .as(JavaArchive.class);
+
+        return ShrinkWrap.create(WebArchive.class, "interceptorFtCdi.war")
+            .addAsLibrary(testJar);
+    }
+
+    @Test
+    public void testEjbAsync() throws InterruptedException, ExecutionException {
```

Comment:
This test was originally written just to test that async works at all with EJBs. It doesn't actually use interceptors at all.

We need to take it out, or change it so that it's testing async with interceptors (e.g. testing that the interceptors fire in the correct order and on the correct threads)

Reply:
this was fixed and the test for the async annotation was added.